src/Aspire.Hosting.Testing/DistributedApplicationExtensions.cs

```diff
@@ -98,6 +100,15 @@ private static string GetEndpointUriStringCore(DistributedApplication app, strin
         return endpoint.Url;
     }
 
+    static void ThrowIfNotStarted(DistributedApplication app)
+    {
+        var lifetime = app.Services.GetRequiredService<IHostApplicationLifetime>();
+        if (!lifetime.ApplicationStarted.IsCancellationRequested)
+        {
+            throw new InvalidOperationException("The application has not started.");
```

Comment:
These message should be in resources now so we can localize. Can we make this a little more descriptive, e.g. "The application must be started before resolving endpoints or connection strings."

Reply:
Pushed update. PTAL. I thought we stopped localizing exception messages.